pkg/kubectl/cmd/util/factory.go

```diff
@@ -371,7 +421,7 @@ func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
 	mapper = NewShortcutExpander(mapper, discoveryClient)
 
 	// wrap with output preferences
-	cfg, err := f.clients.ClientConfigForVersion(nil)
+	cfg, err := ccf.ClientCache().ClientConfigForVersion(nil)
```

Comment:
This ought to negotiate the same regardless of OpenShift or kube and every future openshift version will be API discoverable.